src/main/java/com/fauna/response/wire/MultiByteBufferInputStream.java

```diff
@@ -0,0 +1,55 @@
+package com.fauna.response.wire;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+/**
+ * Joins a list of byte buffers to make them appear as a single input stream.
+ *
+ * The reset method is supported, and always resets to restart the stream at the beginning of the first buffer,
+ * although markSupported() returns false for this class.
+ */
+public class MultiByteBufferInputStream extends InputStream {
```

Comment:
Based on what @erickpintor mentioned about emitting /n between events, should we be implementing readLine() in this wrapper instead?

Reply:
I think readline would fail if you return an unescaped `\n` in your data.